### ISSUE TITLE
Integration tests stall after launching minecraft

### ISSUE THREAD

**Issue by DaveyBiggers** [Microsoft/malmo#451]
Minecraft/launch_minecraft_in_background.py calls launchClient.sh/.bat
The integration tests use this to launch the required number of Minecraft clients before testing.
There are a number of issues with this:
a) The requested ports were not being passed to launchClient
b) The call to launchClient assumes the cwd is the Minecraft dir - this is usually not true when running via ctest
c) ctest waits until all _child_ processes have terminated before continuing. On certain systems this causes the tests to pause until Minecraft is closed (which obviously defeats the whole point of running it in the first place).

a) and b) are easy to remedy.
c) is proving problematic.

**Comment by DaveyBiggers** [comment]
The child process issue turns out to be a red-herring. CTest does indeed wait until all child processes have finished, here:

```
      /* Wait for all processes to terminate.  */
      while(wait(&procStatus) >= 0 || errno != ECHILD)
        {
        }
```
but in our case the calls to subprocess.Popen in launch_minecraft_in_background.py are correctly disassociating the spawned processes, as can be verified using `pstree`.

CTest still hangs until the java processes are killed, but it turns out to be because CTest is also waiting for the stdin, stdout and stderr pipes to close, and they are being inherited (and hence kept open) by the child processes. The solution appears to be to force the creation of new pipes for the children, by adding `stdin = subprocess.PIPE, stdout = subprocess.PIPE, stderr = subprocess.PIPE` to the Popen calls.

This seems to fix things on Ubuntu 16.04 in Azure - still need to test on the other OSs.

**Comment by DaveyBiggers** [comment]
Fixed in #454 for Ubuntu, Fedora and Windows - am closing for now. Will reopen if there are problems with other OSs.
